include/cache.php

```diff
@@ -46,43 +50,70 @@ public function switch_blog( $new_blog_id ) {
 	}
 
 	/**
-	 * Add a value in cache.
+	 * Adds a value in cache.
 	 *
 	 * @since 1.7
+	 * @since 3.6 Returns the cached value.
 	 *
 	 * @param string $key  Cache key.
 	 * @param mixed  $data The value to add to the cache.
-	 * @return void
+	 * @return mixed
+	 *
+	 * @phpstan-param non-empty-string $key
+	 * @phpstan-param TCacheData $data
+	 * @phpstan-return TCacheData
 	 */
-	public function set( $key, $data ) {
+	public function set( string $key, $data ) {
 		$this->cache[ $this->blog_id ][ $key ] = $data;
+		return $data;
 	}
 
 	/**
-	 * Get value from cache.
+	 * Returns value from cache.
 	 *
 	 * @since 1.7
 	 *
 	 * @param string $key Cache key.
 	 * @return mixed
+	 *
+	 * @phpstan-param non-empty-string $key
+	 * @phpstan-return TCacheData|false
 	 */
-	public function get( $key ) {
+	public function get( string $key ) {
 		return isset( $this->cache[ $this->blog_id ][ $key ] ) ? $this->cache[ $this->blog_id ][ $key ] : false;
 	}
 
 	/**
-	 * Clean the cache (for this blog only).
+	 * Cleans the cache (for this blog only).
 	 *
 	 * @since 1.7
 	 *
-	 * @param string $key Cache key.
+	 * @param string $key Optional. Cache key. An empty string to clean the whole cache for the current blog.
+	 *                    Default is an empty string.
 	 * @return void
 	 */
-	public function clean( $key = '' ) {
-		if ( empty( $key ) ) {
+	public function clean( string $key = '' ) {
+		if ( '' === $key ) {
 			unset( $this->cache[ $this->blog_id ] );
 		} else {
 			unset( $this->cache[ $this->blog_id ][ $key ] );
 		}
 	}
+
+	/**
+	 * Generates and returns a "unique" cache key, depending on `$data` and prefixed by `$prefix`.
+	 *
+	 * @since 3.6
+	 *
+	 * @param string              $prefix String to prefix the cache key.
```

Comment:
I would require `$prefix` to be non-empty, otherwise we could have an empty cache key, couldn't we?

Reply:
No but you highlight something incorrect with what I've done.
I did this:
```
@phpstan-return ( $prefix is non-empty-string ? non-empty-string : string )
```
But this is incorrect, because even if `$prefix` is an empty string, the returned cache key will not be an empty string, thanks to `md5()`.

I would keep the possibility to have `$prefix` as an empty string because we may want to generate several keys with the same md5. Ex:
```php
$key            = $this->cache->get_unique_key( '', $data );
$foobar_key     = 'foobar_' . $key;
$foobar_baz_key = 'foobar_baz_' . $key;
```

But maybe we may prefer to enforce `$prefix` as a non empty string to prevent bugs where `$prefix` is empty while we expect it not to be?
What do you think?

Reply:
Forcing `$prefix` to non-empty is safer IMHO. Also, you can still do:
```php
$foobar_key = $this->cache->get_unique_key( 'foobar_', $data );
$foobar_baz_key = $this->cache->get_unique_key( 'foobar_baz_', $data );
```
And you get the same result, right? Or am I missing something 🤔

Reply:
Done in [150d298](https://github.com/polylang/polylang/pull/1403/commits/150d298b78a398c7e390d7be93e513aa44028f21).